src/lints/enum_no_repr_variant_discriminant_changed.ron

```diff
@@ -0,0 +1,78 @@
+    description: "A public enum's variant had its discriminant changed from its previous value.",
+    reference: Some("A public enum's variant had its discriminant changed from its previous value, which breaks implementors relying on the discriminant value."),
+    required_update: Major,
+    lint_level: Deny,
+    reference_link: Some("https://doc.rust-lang.org/reference/items/enumerations.html#assigning-discriminant-values"),
+    query: r#"
+    {
+        CrateDiff {
+            baseline {
+                item {
+                    ... on Enum {
+                        visibility_limit @filter(op: "=", value: ["$public"]) @output
+                        enum_name: name @output @tag
+
+                        attribute @optional {
+                            content {
+                                base @filter(op: "!=", value: ["$repr"])
+                            }
+                        }
+
+                        importable_path {
+                            path @output @tag
+                            public_api @filter(op: "=", value: ["$true"])
+                        }
+
+                        variant {
+                            variant_name: name @output @tag
+                            discriminant @optional {
+                                old_value: value @output @tag
+                            }
+                        }
+                    }
+                }
+            }
+            current {
+                item {
+                    ... on Enum {
+                        visibility_limit @filter(op: "=", value: ["$public"])
+                        name @filter(op: "=", value: ["%enum_name"])
+
+                        attribute @optional {
+                            content {
+                                base @filter(op: "!=", value: ["$repr"])
+                            }
+                        }
+
+                        importable_path {
+                            path @filter(op: "=", value: ["%path"])
+                            public_api @filter(op: "=", value: ["$true"])
+                        }
+
+                        variant {
+                            name @filter(op: "=", value: ["%variant_name"])
+                            discriminant {
+                                new_value: value @output @filter (op: "!=", value: ["%old_value"])
+                            }
+                        }
+
+                        span_: span @optional {
+                            filename @output
+                            begin_line @output
+                        }
+                    }
+                }
+            }
+        }
+    }"#,
+    arguments: {
+        "public": "public",
+        "repr": "repr",
+        "true": true,
+    },
+    error_message: "The public enum's variant had its discriminant changed from its previous value. This breaks implementors that relied on the discriminant value.",
```

Comment:
Let's make sure to be very specific about what broke, and precise with terminology. Enums aren't traits so they don't have implementors.
```suggestion
    error_message: "The enum's variant had its discriminant value change. This breaks downstream code that used its value via a numeric cast like `as isize`.",
```